fe/fe-core/src/main/java/com/starrocks/analysis/ArithmeticExpr.java

```diff
@@ -141,8 +141,26 @@ private static Type nonDecimalToDecimal(Type type) {
         }
     }
 
+    public static void getAddReturnTypeOfDecimal(TypeTriple triple, ScalarType lhsType, ScalarType rhsType) {
+        final int lhsPrecision = lhsType.getPrecision();
+        final int rhsPrecision = rhsType.getPrecision();
+        final int lhsScale = lhsType.getScalarScale();
+        final int rhsScale = rhsType.getScalarScale();
+
+        int delataScale = Math.abs(lhsScale - rhsScale);
```

Comment:
typo, delta

decimal(p1, s1)+decimal(p2,s2) = decimal(max(p1-s1, p2-s2)+1+max(s1,s2), max(s1, s2))

Reply:
I will it as a better algorithm